src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -820,14 +820,16 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
 
                 lock (pathList.SyncRoot)
                 {
-                    foreach (DictionaryEntry entry in pathList)
```

Comment:
This change is faster (on my machine), presumably because the extra interface call is cheaper than a box allocation per entry.

```
               Method |     Mean |    Error |   StdDev |
--------------------- |---------:|---------:|---------:|
      DictionaryEntry | 213.9 ns | 3.112 ns | 2.759 ns |
 DictionaryEnumerator | 151.4 ns | 2.425 ns | 2.269 ns |
```

```c#
public class PathListBenchmarks
{
    private readonly PathList _list = new PathList();

    public PathListBenchmarks()
    {
        var collection = new CookieCollection();
        for (int i = 0; i < 10; i++)
        {
            _list[i.ToString()] = collection;
        }
    }

    [Benchmark]
    public void DictionaryEntry()
    {
        string path;
        CookieCollection cc;

        foreach (DictionaryEntry entry in _list)
        {
            path = (string)entry.Key;
            cc = (CookieCollection)entry.Value;
        }
    }

    [Benchmark]
    public void DictionaryEnumerator()
    {
        string path;
        CookieCollection cc;

        IDictionaryEnumerator e = _list.GetEnumerator();
        while (e.MoveNext())
        {
            path = (string)e.Key;
            cc = (CookieCollection)e.Value;
        }
    }
}
```